File: mian.py


# 1. Function to reverse a string
def reverse_string(s):
    return s[::-1]

# 2. Function to calculate the factorial of a number
def factorial(n):
    if n <= 1:
        return 1
    return n * factorial(n - 1)

# 3. Function to check if a number is prime
def is_prime(num):
    if num <= 1:
        return False
    for i in range(2, int(num ** 0.5) + 1):
        if num % i == 0:
            return False
    return True


# Using the reverse_string function
reversed_str = reverse_string("Hello World")
print("Reversed string:", reversed_str)  # Output: "dlroW olleH"
# Using the factorial function
fact = factorial(5)
print("Factorial of 5:", fact)  # Output: 120
# Using the is_prime function
prime_check = is_prime(7)
print("Is 7 prime?", prime_check)  # Output: True
prime_check = is_prime(10)
print("Is 10 prime?", prime_check)  # Output: False

File: mian.js


// 1. Function to reverse a string
function reverseString(str) {
    return str.split('').reverse().join('');
}

// 2. Function to calculate the factorial of a number
function factorial(n) {
    if (n <= 1) return 1;
    return n * factorial(n - 1);
}

// 3. Function to check if a number is prime
function isPrime(num) {
    if (num <= 1) return false;
    for (let i = 2; i <= Math.sqrt(num); i++) {
        if (num % i === 0) return false;
    }
    return true;
}


// Using the reverseString function
let reversed = reverseString("Hello World");
console.log("Reversed string:", reversed);
// Using the factorial function
let fact = factorial(5);
console.log("Factorial of 5:", fact);
// Using the isPrime function
let primeCheck = isPrime(7);
console.log("Is 7 prime?", primeCheck);
primeCheck == isPrime(10);
console.log("Is 10 prime?", primeCheck);